resources/params.mjs

```diff
@@ -73,9 +71,11 @@ class Params {
 }
 
 export const defaultParams = new Params();
+Object.freeze(defaultParams.viewport);
```

Comment:
I mostly wanted to guard against accidental modifications of the params objects.
Maybe you can freeze the params object if there are no searchParams (aka the default mode?):

```
    ...
    constructor(searchParams = undefined) {
        if (searchParams)
            this._copyFromParams(searchParams);
        if (!searchParams || searchParams.size == 0) {
            Object.freeze(this.viewport);
            Object.freeze(this);
       }
    }
    ...
```

Reply:
I thought this wouldn't work initially (i.e. if there are no searchParams at load and then you start wanting to configure). Though with how it's currently built there must by at least one (`developerMode`) to get into this configuration mode. It seems a little coincidental but I don't have a strong preference here so we can update that with a comment.

Reply:
Hm, this change was leading to readonly errors when toggling controls locally. I think I will go ahead with the current change where we explicitly freeze only the defaultParams object and then revisit this along with the URL encoding moving into params object.